### DISCUSSION TITLE
[Help] Problem with linking params in web

### DISCUSSION THREAD

**Discussion by moyolvera** [react-navigation/react-navigation#10360] [Q&A] [answered]
Hi everyone, I'm trying to configure linking on my react native app for web.

I'm using:
Expo SDK 44
react-navigation: 6.0.8

This is my current config:

```
const linking = {
  prefixes: [],
  config: {
    screens: {
      Home: 'invitation/:token?',
      Details: 'details'
    }
  }
};
```

And this is my stack:

```
function Navigator() {
  return (
    <NavigationContainer linking={linking}>
      <Stack.Navigator screenOptions={SCREEN_OPTIONS}>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Details" component={DetailsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
```

When running locally:
  - If i go to `localhost`, it redirects me to `localhost/invitation` which is good
  - if i go to `localhost/invitation?token=thisisatoken` i receive the param `token` fine which is fine also
  - if i go to `localhost/details` it loads the page just fine
  
  ### The problem is when running the build version
  
  - If i go to `weburl`, it redirects me to `localhost/invitation` which is good
  - **if i go to `weburl/invitation?token=thisisatoken` it sends me to a 404 page** 🔴 
  - **if i go to `weburl/details` it sends me to a 404 page** 🔴
  
I cant find why it works while running the development version but not on the production build, any help will be so awesome

Thank you for your time :)

**Comment by satya164** [comment, marked as answer]
You need to configure your hosting provider to serve the same main HTML file regardless of URL so that your code can be loaded to handle the URL. Search for any docs related to SPA or client side routing related to your hosting provider.

**Reply by moyolvera** [reply]
> You need to configure your hosting provider to serve the same main HTML file regardless of URL so that your code can be loaded to handle the URL. Search for any docs related to SPA or client side routing related to your hosting provider.

Thank you sir, that solved my issue!

In case someone using firebase hosting encounters this behavior i only added this to my firebase.json

```
"hosting": {
    ...
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  }
  ```
